src/AnalyticsEventReporter.ts

```diff
@@ -18,13 +18,6 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
    * @param payload - (optional) desired event values to report
    */
   constructor(config: AnalyticsConfig, payload?: EventPayload) {
-    const apiKeyIsSet = config.key !== undefined;
-    const bearerTokenIsSet = config.bearer !== undefined;
-    const configIsValid =
-      (apiKeyIsSet || bearerTokenIsSet) && !(apiKeyIsSet && bearerTokenIsSet);
-    if (!configIsValid) {
```

Comment:
I think you need to still validate that the authorization and authorizationKey are defined. Someone not using typescript won't get the type checking so could still not set the values at all.